### PR TITLE
Feat/search

### PR DESCRIPTION
**Pull request by mingard**
Some improvements to the basic search method

### CODE REVIEW

**Comment by mingard** [comment]
@jimlambie @eduardoboucas I'll put some docs together and talk you through this shortly

**Comment by mingard** [comment]
Ah yes of course. Will do

**Comment by mingard** [comment]
@jimlambie done

File: dadi/lib/controller/index.js [hunk 1]
```diff
@@ -33,11 +34,44 @@ function ApiError (status, code, message, title) {
...
+  this.model.get(query, queryOptions, done, req)
+}
+
+const prepareQueryOptions = function (options, modelSettings) {
   var response = { errors: [] }
   var queryOptions = {}
   var settings = modelSettings || {}
 
+  if (options.search) {
+    let searchQuery = parseQuery(options.search)
```
Comment (eduardoboucas) [inline dadi/lib/controller/index.js]: The build is breaking on versions 4 and 5 of Node because there's a missing strict mode header in this file.

Comment (mingard) [inline dadi/lib/controller/index.js]: Fixed

**Comment by mingard** [comment]
@jimlambie @eduardoboucas I'll document this shortly. You can wait to merge until then.

**Pull request closed without merging**